blockchain/tx_pool.go

```diff
@@ -1717,17 +1730,27 @@ func (as *accountSet) add(addr common.Address) {
 // peeking into the pool in TxPool.Get without having to acquire the widely scoped
 // TxPool.mu mutex.
 type txLookup struct {
-	all  map[common.Hash]*types.Transaction
-	lock sync.RWMutex
+	all   map[common.Hash]*types.Transaction
+	slots int
+	lock  sync.RWMutex
 }
 
 // newTxLookup returns a new txLookup structure.
 func newTxLookup() *txLookup {
+	slotsGauge.Update(int64(0))
 	return &txLookup{
 		all: make(map[common.Hash]*types.Transaction),
 	}
 }
 
+// Slots returns the current number of slots used in the lookup.
+func (t *txLookup) Slots() int {
```

Comment:
Seems unsafe because the returned integer may not be synched one if another thread modifies it at the same time. But, we may overlook it for now as its side-effect causes only wrong dropping. Any comment?

Reply:
@hyunsooda Does the lock protects the section, doesn't it? Is there any places which is not protected?

Reply:
Uh, the returned value of `Slots()` does not have a [lifetime](https://doc.rust-lang.org/book/ch10-03-lifetime-syntax.html) (the reference is Rust-based, but I believe the concept is not different) as it is not assigned to a variable. Thus, it seems not unsafe. I concerned if the value is assigned to the variable, then the variable (primitive value(integer))may have unsynched one with other contexts that may modifies it at the same time.

Reply:
What happens if there's lifetime?

Reply:
As we discussed in offline, it won't happen fundamentally. Only `Txpool.add` function calls the `Slots` function to access `t.slots`, but this `add` function would not be executed concurrently.